tests.py

```diff
@@ -0,0 +1,105 @@
+import unittest
+from zoo_project import Cage, Animal, Zoo
+
+
+class TestZooCreation(unittest.TestCase):
+    """Just creates a Zoo and checks the name is the same one we used at creation time.
+    """
+
+    def test_zoo(self):
+        zoo = Zoo("London Zoo")
+        self.assertEqual(zoo.name, 'London Zoo')
+
+
+class TestCageCreation(unittest.TestCase):
+    """Creates a cage
+    """
+
+    def test_cage(self):
+        cage = Cage()
+
+
+class TestAnimalCreation(unittest.TestCase):
+    """Creates an animal and checks that the name corresponds to the one we used at creation time.
+    """
+
+    def test_animal(self):
+        freddie_the_lion = Animal("Freddie", "Lion")
+        self.assertEqual(freddie_the_lion.animal_name, 'Freddie')
+
+
+class TestAddAnimalToCage(unittest.TestCase):
+    """Creates an animal, adds it to the cage and checks that the cage contains an animal with the right name in the text.
+    """
+
+    def test_add_animal_to_cage(self):
+        freddie_the_lion = Animal("Freddie", "Lion")
+        cage = Cage()
+        cage.add_animal(freddie_the_lion)
+        self.assertEqual(str(cage.animals[0]), 'Freddie the Lion')
```

Comment:
A more 'pure' test would be:

    self.assertEqual(cage.animals, [freddie_the_lion])